### PR TITLE
BOAC-179, Google Analytics > track CAS login, view cohort and view student

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-179

* devAuth login is not tracked
* casLogin could be tracked server-side but it's not worth the effort, thus the minor hack.
* Service Lead can decide on other important events to track, if any. (Get out the Magic Eight Ball.)

### CODE REVIEW

File: boac/static/app/cohort/cohortController.js [hunk 1]
```diff
@@ -188,6 +188,8 @@
               var teamCodes = cohort.code ? [ cohort.code ] : _.map(cohort.teams, 'code');
               $scope.search.count.selectedTeams = teamCodes.length;
               refreshTeamsFilter(teamCodes);
+              // Track view event
+              googleAnalyticsService.track('cohort', 'view', cohort.label || cohort.name, code);
```
Comment (johncrossman) [inline boac/static/app/cohort/cohortController.js]: Google documentation says final arg above must be an int (see js doc on `track` function). Team cohorts have three character (alpha) `code`. How will Google handle it?!

File: boac/static/app/routes.js [hunk 2]
```diff
@@ -34,12 +34,7 @@
       })
       .state('manageCohorts', {
         url: '/cohorts/manage',
-        templateUrl: '/static/app/cohort/manage.html',
-        controller: 'ManageCohortsController'
-      })
-      .state('manageCohort', {
-        url: '/cohorts/manage/{cohort_id}',
-        templateUrl: '/static/app/cohort/manage.html',
+        templateUrl: '/static/app/cohort/manageCohorts.html',
```
Comment (johncrossman) [inline boac/static/app/routes.js]: `/cohorts/manage/{cohort_id}` is obsolete.

**Comment by johncrossman** [comment]
I want to make it couple amendments here. I'll reopen PR soon.

File: boac/static/app/student/studentController.js [hunk 3]
```diff
@@ -2,12 +2,17 @@
...
     var loadAnalytics = authService.authWrap(function() {
+      var uid = $stateParams.uid;
+
       $scope.student.isLoading = true;
-      studentFactory.analyticsPerUser($stateParams.uid).then(function(analytics) {
+      studentFactory.analyticsPerUser(uid).then(function(analytics) {
         $scope.student = analytics.data;
+        // Track view event
+        var preferredName = student.sisProfile && student.sisProfile.preferredName;
+        googleAnalyticsService.track('student', 'view-profile', preferredName, parseInt(uid, 10));
```
Comment (johncrossman) [inline boac/static/app/student/studentController.js]: After some thought, I decided that 'student' should be the subject of tracked event (above).

**Comment by paulkerschen** [comment]
OUTLOOK GOOD.

**Pull request merged**